File: Testing/Frontend/DefaultLogin/src/test/java/com/Tests/LoginToWelcomeTest.java
package com.Tests;

import com.Page.Login;
import com.Page.Welcome;

import org.junit.jupiter.api.*;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class LoginToWelcomeTest {

    public WebDriver driver;
    public WebDriverWait wait;

    public String serverUrl;

    public String clientId;

    public String realm;

    public String chromeDriverPath;

    public String username;

    public String password;

    @BeforeAll
    public void setUp(){
        this.serverUrl = System.getenv("SERVER_URL");
        this.clientId = System.getenv("CLIENT_ID");
        this.realm = System.getenv("REALM");
        this.username = System.getenv("TEST_USERNAME");
        this.password = System.getenv("TEST_PASSWORD");
        this.chromeDriverPath = System.getenv("CHROMEDRIVER_PATH");

        ChromeOptions options = new ChromeOptions();

        options.addArguments("--remote-allow-origins=*");
        options.addArguments("--window-size=1920,1080","--ignore-certificate-errors","--no-sandbox", "--disable-dev-shm-usage");
        driver = new ChromeDriver(options);
        wait = new WebDriverWait(driver, Duration.ofMillis(2000));
    }

    @Test
    public void LoginToWelcome() throws InterruptedException {
        Login loginPage = new Login(driver , wait, chromeDriverPath);
        loginPage.setup();
        loginPage.url(serverUrl+"/realms/"+realm+"/protocol/openid-connect/auth?response_type=code&client_id="+clientId);
        try {
            Thread.sleep(10000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        loginPage.InsertUsername(username);
        loginPage.InsertPassword(password);
        loginPage.ClickLogin();
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        Welcome welcomePage = new Welcome(driver, wait, chromeDriverPath);
        System.out.println("Expected: Welcome to Keycloak\nGiven: " + welcomePage.getWelcomeHeader());
        Assertions.assertEquals("Welcome to Keycloak" , welcomePage.getWelcomeHeader());

    }

    @AfterAll
    public void close(){
        driver.quit();
    }

}

File: Testing/Frontend/SeleniumJs/LoginToWelcome.test.mjs
import { Builder, By, Key, until } from 'selenium-webdriver';
import { expect } from 'chai';
import * as fs from 'fs';
import chrome from 'selenium-webdriver/chrome.js';

describe('Login to Welcome Test', function () {
  this.timeout(60000);

  let driver;
  const screenshotsDir = './screenshots';

  const serverUrl = process.env.SERVER_URL;
  const realm = process.env.REALM;
  const clientId = process.env.CLIENT_ID;
  const username = process.env.TEST_USERNAME;
  const password = process.env.TEST_PASSWORD;
  const browser = process.env.BROWSER || 'chrome';

  if (!serverUrl || !realm || !clientId || !username || !password) {
    throw new Error('Missing required environment variables');
  }

  if (!fs.existsSync(screenshotsDir)) {
    fs.mkdirSync(screenshotsDir);
  }

  before(async function () {
    const options = new chrome.Options();
    options.addArguments(
      '--remote-allow-origins=*',
      '--window-size=1920,1080',
      '--ignore-certificate-errors',
      '--no-sandbox',
      '--disable-dev-shm-usage'
    );

    driver = await new Builder()
      .usingServer('http://localhost:4444/wd/hub')
      .forBrowser('chrome')
      .setChromeOptions(options)
      .build();
      console.log("Driver set")
  });

  after(async function () {
    if (driver) {
      await driver.quit();
    }
  });

  afterEach(async function () {
    if (driver && this.currentTest?.state === 'failed') {
      const encodedString = await driver.takeScreenshot();
      const filename = `${screenshotsDir}/${this.currentTest.title.replace(/[^a-z0-9]/gi, '_').toLowerCase()}.png`;
      fs.writeFileSync(filename, encodedString, 'base64');
    }
  });

  it('should log in and reach the welcome page', async function () {
    const loginUrl = `${serverUrl}/realms/${realm}/protocol/openid-connect/auth?response_type=code&client_id=${clientId}`;
    await driver.get(loginUrl);
    console.log("Login URL set")
    await driver.sleep(10000); // Optional wait for page load
    console.log("Page loaded")
    const usernameField = By.id('username');
    const passwordField = By.id('password');
    const loginButton = By.id('kc-login');

    await driver.wait(until.elementLocated(usernameField), 5000);
    await driver.findElement(usernameField).sendKeys(username);
    await driver.findElement(passwordField).sendKeys(password);
    await driver.findElement(loginButton).click();

    await driver.sleep(10000); // Wait for redirect

    const welcomeHeader = By.css('body > div > div > div > div.welcome-header > h1');
    await driver.wait(until.elementLocated(welcomeHeader), 5000);
    const headerText = await driver.findElement(welcomeHeader).getText();

    console.log(`Expected: Welcome to Keycloak\nGiven: ${headerText}`);
    expect(headerText).to.equal('Welcome to Keycloak');
  });
});
